### PR TITLE
added client and header timeout.

### PR DESCRIPTION
**Pull request by csalvo9**
This PR adds 2 new methods:

SetHeaderTimeout(int)
This sets the response header timeout (in seconds) for the http transport.

SetClientTimeout(int)
This sets the command timeout (in seconds). Default is 30 seconds (same as before).

These new timeout values are set in callAPI() during transport / http client init.

This PR also changes the call to callAPI from:
     resp, err := callAPI("jsonrpc", url, payload, cli.username, cli.password, .......)
to:
     resp, err := cli.callAPI("jsonrpc", url, payload)

### CODE REVIEW

**Comment by greenpau** [comment]
@csalvo9 , thank you for contributing! 👍 

Please change 

```
func callAPI(contentType string, url string, payload []byte, username, password string, secure bool) ([]byte, error) {
```

to:

```
func (cli *Client) callAPI(contentType string, url string, payload []byte)
```

This way, instead of:

```
resp, err := callAPI("jsonrpc", url, payload, cli.username, cli.password, cli.secure, cli.headerTimeout, cli.clientTimeout)
```

You will do:

```
resp, err := cli.callAPI("jsonrpc", url, payload)
```

File: pkg/client/client.go [hunk 1]
```diff
@@ -191,21 +205,28 @@ func (cli *Client) SetSecure() error {
 	return nil
 }
 
-func callAPI(contentType string, url string, payload []byte, username, password string, secure bool) ([]byte, error) {
+func callAPI(contentType string, url string, payload []byte, username, password string, secure bool, headerTimeout time.Duration, clientTimeout time.Duration) ([]byte, error) {
```
Comment (greenpau) [inline pkg/client/client.go:208]: @csalvo9, please change this function.

**Review by greenpau — CHANGES_REQUESTED** [review]

File: pkg/client/client.go (hunk 1 above)
Comment (csalvo9) [reply pkg/client/client.go]: updated callAPI.

**Review by greenpau — APPROVED** [review]
@csalvo9 , LGTM 👍

**Comment by greenpau** [comment]
@csalvo9 , if you have energy, please see if you could resolve https://github.com/greenpau/go-cisco-nx-api/issues/6?

**Pull request merged**